internal/oci/containerd.go

```diff
@@ -227,10 +223,12 @@ func (c *Containerd) GetBlob(ctx context.Context, dgst digest.Digest) ([]byte, s
 	if err := json.Unmarshal(b, &ud); err != nil {
 		return nil, "", err
 	}
-	if ud.MediaType == "" {
-		return nil, "", fmt.Errorf("blob manifest cannot be empty")
+	if ud.MediaType != "" {
+		return b, ud.MediaType, nil
 	}
-	return b, ud.MediaType, nil
+	// Media type is not a required field in any file. We need a fallback method if the field is not set.
+	logr.FromContextOrDiscard(ctx).Info("using Containerd fallback method to determine media type", "digest", dgst.String())
```

Comment:
The last challenge is figuring out what our fallback method should look like. We have a couple of options and none of them are great. The issue is that the media type is set in the "parent descriptor" and we don't have a good method of walking backwards.

Currently the options that I have come up with:

* Keep track of all digests and media types seen when iterating images in a hash map.
* Use Containerd content labels to filter and find parent content.
* Walk all content until parent content is found.
* Use labels on content to do a best guess of content type.
* Attempt to parse content into different structs until one works.